src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -2559,6 +2559,21 @@ public static void DCS_NonGenericDictionaryOfStringInt32()
 
     #endregion
 
+    [Fact]
+    public static void BasicRoundTripResolveDTOTypes()
+    {
+        ObjectContainer instance = new ObjectContainer(new DTOContainer());
+        XmlObjectSerializer serializer = new DataContractSerializer(typeof(ObjectContainer), null, null, null, int.MaxValue, false, false, new DTOResolver());
+        using (MemoryStream ms = new MemoryStream())
+        {
```

Comment:
You can use SerializeAndDeserialize method and provide a serializerFactory instead of creating serializer and stream on your own.